src/scripts/main.js

```diff
@@ -1,3 +1,282 @@
 'use strict';
 
-// write your code here
+const messegLose = document.querySelector('.message-lose');
+const messegWin = document.querySelector('.message-win');
+const messegStart = document.querySelector('.message-start');
+const Startbutton = document.querySelector('.start');
+const Restarttbutton = document.querySelector('.restart');
+const scoreTable = document.querySelector('.game-score');
+const tiles = document.getElementsByClassName('tile');
+
+const rows = 4;
+const columns = 4;
+let score = 0;
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+let stopGame = false;
+
+Startbutton.addEventListener('click', () => {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const tile = document.createElement('div');
+      const num = board[r][c];
+
+      tile.id = r.toString() + '-' + c.toString();
+
+      upDateTile(tile, num);
+      document.getElementById('board').append(tile);
+    }
+  }
+
+  Startbutton.classList.add('hidden');
+  messegStart.classList.add('hidden');
+  Restarttbutton.classList.remove('hidden');
+
+  addNum();
+  addNum();
+});
+
+Restarttbutton.addEventListener('click', function restartGame() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  const tileArr = Array.from(tiles);
+
+  tileArr.forEach((tile) => {
+    tile.innerText = '';
+    tile.className = 'tile';
+  });
+
+  score = 0;
+  scoreTable.innerHTML = score;
+  messegLose.classList.add('hidden');
+  messegWin.classList.add('hidden');
+  addNum();
+  addNum();
+});
+
+function endGame() {
+  if (gameOver()) {
+    stopGame = false;
+    messegLose.classList.remove('hidden');
+  }
+}
+
+function empty() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function addNum() {
+  if (!empty()) {
+    return true;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      const num = Math.random() < 0.01 ? 4 : 2; // Fix the missing parentheses
```

Comment:
Remove comment. Try to write obvious code with meaningful variable and function naming